Lib/email/utils.py

```diff
@@ -48,6 +48,45 @@
 specialsre = re.compile(r'[][\\()<>@,:;".]')
 escapesre = re.compile(r'[\\"]')
 
+_EMAIL_CONFIG_FILE = "/etc/python/email.cfg"
+
+
+def _use_strict_email_parsing():
+    """Returns True if strict email parsing is not disabled by
+    config file or env variable.
+    """
+    global _cached_strict_addr_parsing
+
+    try:
+        return _cached_strict_addr_parsing
+    except NameError:
+        disabled = bool(os.environ.get("PYTHON_EMAIL_DISABLE_STRICT_ADDR_PARSING"))
+        if disabled:
+            _cached_strict_addr_parsing = False
+            return _cached_strict_addr_parsing
```

Comment:
You may move the code into a new `_use_strict_email_parsing_impl()` function, and only keep the code managing the cache here. Something like:

```py
_cached_strict_addr_parsing = None

def _use_strict_email_parsing():
    """"...doc..."""
    global _cached_strict_addr_parsing
    if _use_strict_email_parsing is None:
        _use_strict_email_parsing = _use_strict_email_parsing_impl()
    return _use_strict_email_parsing
```

Reply:
Do you think the current implementation is too complex so it needs another layer? Or is there any other benefit of the proposed solution I don't see?

Reply:
It's just a coding style suggestion, IMO it makes the code easier to read/maintain. It's up to you.

Reply:
I agree, I like the new implementation - nice separation of concerns.